solutions/cses-1736.mdx

```diff
@@ -0,0 +1,151 @@
+---
+id: cses-1736
+source: CSES
+title: Polynomial Queries
+author: Albert Ye
+---
+
+## Solution
+**Time Complexity: $\mathcal{O}(n\log n)$**
+
+Despite the strange operations we can still use a segment tree with lazy propagation
+to solve this problem.
+
+The range queries are vanilla lazy propagation, so we should focus more on the updates.
+
+We can split the "update" operation into two operations. One would be a basic range addition,
+and the other would be an incremental addition. Therefore, for each index $a_i$, we will need
+to keep track of two points: the total sum $s_i$ given only the range sums without increments,
+and a value $t_i$ which contains the sum of all increments from the incremental steps.
+
+When we update, we want to keep two lazy values: one (`lz`) stores the unapplied change in $s_i$,
+much like regular lazy prop,
+and the other (`lz2`) stores the unapplied change from the incremental sum $t_i$.
+When we propagate, the value of `lz` for
+a node `nd` containing segment $[i, j]$ when updating $[l, r]$ can be found by calculating $i-l$.
+When we update increments for any node `nd` we always add 1 to `lz2[nd]` as per the 
+problem statement.
```

Comment:
I think the easier solution to explain is to have one segment tree that stores:
1) a[i]
2) b[i]*i
with range updates of a[i] and b[i].

Anyways, to explain the solution here, probably clarify what the two update operations are
1) Range addition: a[i] += x for some constant x
2) Incremental: a[i] += (i-l)*x for some constant x, for everything in the segment (l, r)

And in the segment tree every segment that needs to be updated for each input query can be handled by a combination of (1) and (2). (1) to offset position and (2) to handle the incremental nature

imo describing `lz` and `lz2` doesn't help understand w/o looking at the code (you can add comments to explain them better in the code tho)

Reply:
oops i totally forgot to specify the updates